charts/k8s-monitoring/charts/feature-profiling/templates/_ebpf.tpl

```diff
@@ -32,6 +33,19 @@ discovery.relabel "ebpf_pods" {
     regex = "Succeeded|Failed|Completed"
     action = "drop"
   }
+{{- if eq .Values.ebpf.targetingScheme "annotation" }}
+  rule {
+    source_labels = [{{ $scrapeAnnotation | quote }}]
+    regex         = "true"
+    action        = "keep"
+  }
+{{- else }}
+  rule {
+    source_labels = [{{ $scrapeAnnotation | quote }}]
+    regex         = "false"
+    action        = "drop"
+  }
+{{- end }}
```

Comment:
I think this would be super helpful to aid debugging once we have profiles ingested. Futhermore it also helps to make sure that if we double collect, we don't have a series conflict.

If you agree with this let's also add it to `_java.tpl` and `_pprof.tpl`



```
          rule {
            replacement = "alloy/pyroscope.ebpf"
            target_label = "profile_source"
          }
```